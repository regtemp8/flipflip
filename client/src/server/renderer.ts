import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
