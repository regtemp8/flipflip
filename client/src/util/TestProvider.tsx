import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
