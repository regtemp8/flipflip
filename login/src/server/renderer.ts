import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
