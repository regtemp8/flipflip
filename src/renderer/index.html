<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
