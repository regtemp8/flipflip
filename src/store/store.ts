import { configureStore } from '@reduxjs/toolkit'
import appSlice from './app/slice'
import audioSlice from './audio/slice'
import constantsSlice from './constants/slice'
import playersSlice from './player/slice'
import captionScriptSlice from './captionScript/slice'
import clipSlice from './clip/slice'
import librarySourceSlice from './librarySource/slice'
import overlaySlice from './overlay/slice'
import playlistSlice from './playlist/slice'
import sceneSlice from './scene/slice'
import sceneGridSlice from './sceneGrid/slice'
import sceneGroupSlice from './sceneGroup/slice'
import tagSlice from './tag/slice'
import videoClipperSlice from './videoClipper/slice'
import scenePickerSlice from './scenePicker/slice'
import captionScriptorSlice from './captionScriptor/slice'
import sceneDetailSlice from './sceneDetail/slice'
import audioLibrarySlice from './audioLibrary/slice'
import sourceScraperSlice from './sourceScraper/slice'

const store = configureStore({
  reducer: {
    app: appSlice,
    audio: audioSlice,
    captionScript: captionScriptSlice,
    clip: clipSlice,
    constants: constantsSlice,
    librarySource: librarySourceSlice,
    overlay: overlaySlice,
    playlist: playlistSlice,
    scene: sceneSlice,
    sceneGrid: sceneGridSlice,
    sceneGroup: sceneGroupSlice,
    tag: tagSlice,
    // components
    players: playersSlice,
    videoClipper: videoClipperSlice,
    scenePicker: scenePickerSlice,
    captionScriptor: captionScriptorSlice,
    sceneDetail: sceneDetailSlice,
    audioLibrary: audioLibrarySlice,
    sourceScraper: sourceScraperSlice
  }
})

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
