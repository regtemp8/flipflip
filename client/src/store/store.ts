/// <reference path="../global.d.ts"/>
import { createReduxStore } from '../server/renderer'

const store = createReduxStore(
  window.flipflipAppStorage,
  window.flipflipConstants
)

export default store
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
