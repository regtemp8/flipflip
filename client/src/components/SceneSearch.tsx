import React, { SyntheticEvent, useState } from 'react'

import {
  Autocomplete,
  AutocompleteChangeDetails,
  AutocompleteChangeReason,
  AutocompleteInputChangeReason,
  TextField,
  type Theme
} from '@mui/material'
import { makeStyles } from 'tss-react/mui'
import { grey } from '@mui/material/colors'

import { useAppDispatch, useAppSelector } from '../store/hooks'
import { selectScenePickerFilters } from '../store/scenePicker/selectors'
import { setScenePickerFilters } from '../store/scenePicker/slice'

const useStyles = makeStyles()((theme: Theme) => ({
  searchSelect: {
    minWidth: 200,
    maxWidth: `calc(100% - ${theme.spacing(7)})`,
    maxHeight: theme.mixins.toolbar.minHeight,
    color: grey[900]
  },
  select: {
    color: grey[900]
  }
}))

export interface SceneSearchProps {
  placeholder: string
}

function SceneSearch(props: SceneSearchProps) {
  const dispatch = useAppDispatch()
  const filters = useAppSelector(selectScenePickerFilters())
  const options = filters.map((filter) => {
    return { label: filter, value: filter }
  })

  const [searchInput, setSearchInput] = useState('')
  const handleInputChange = (
    event: SyntheticEvent<Element, Event>,
    searchInput: string,
    reason: AutocompleteInputChangeReason
  ) => {
    setSearchInput(searchInput)
  }

  const handleChange = (
    event: SyntheticEvent<Element, Event>,
    value: (string | { value: string; label: string })[],
    reason: AutocompleteChangeReason,
    details?: AutocompleteChangeDetails<
      string | { value: string; label: string }
    >
  ) => {
    const filters: string[] = []
    const search = value.map((v) =>
      typeof v === 'string' ? { value: v, label: '' } : v
    )
    for (const s of search) {
      if (
        ((s.value.startsWith('"') || s.value.startsWith('-"')) &&
          s.value.endsWith('"')) ||
        ((s.value.startsWith("'") || s.value.startsWith("-'")) &&
          s.value.endsWith("'"))
      ) {
        filters.push(s.value)
      } else {
        filters.push(...s.value.split(' '))
      }
    }

    const uniqueFilters = filters.filter(
      (value, index, array) => array.indexOf(value) === index
    )
    dispatch(setScenePickerFilters(uniqueFilters))
  }

  const { classes } = useStyles()
  return (
    <Autocomplete
      className={classes.searchSelect}
      multiple
      value={options}
      options={options}
      inputValue={searchInput}
      onInputChange={handleInputChange}
      onChange={handleChange}
      selectOnFocus
      handleHomeEndKeys
      renderOption={(props, option, state) => {
        const { ...optionProps } = props
        return (
          <li key={state.index} {...optionProps}>
            {typeof option === 'string' ? option : option.label}
          </li>
        )
      }}
      filterOptions={(options, params) => {
        const { inputValue } = params
        const missing =
          options.find(
            (option) =>
              (typeof option === 'string' && inputValue === option) ||
              inputValue === (option as { label: string; value: string }).label
          ) == null
        return inputValue !== '' && missing
          ? [
              {
                value: inputValue,
                label: `Search for "${inputValue}"`
              }
            ]
          : []
      }}
      freeSolo
      renderInput={(params) => (
        <TextField
          {...params}
          placeholder={props.placeholder}
          sx={{
            '& .MuiOutlinedInput-root': {
              '&.Mui-focused fieldset': {
                borderColor: 'white'
              }
            }
          }}
        />
      )}
    />
  )
}

;(SceneSearch as any).displayName = 'SceneSearch'
export default SceneSearch
