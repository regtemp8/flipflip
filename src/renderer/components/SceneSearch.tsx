import React, { useState } from 'react'
import clsx from 'clsx'
import CreatableSelect from 'react-select/creatable'

import { type Theme } from '@mui/material'
import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'
import { grey } from '@mui/material/colors'

import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { selectScenePickerFilters } from '../../store/scenePicker/selectors'
import { setScenePickerFilters } from '../../store/scenePicker/slice'

const styles = (theme: Theme) =>
  createStyles({
    searchSelect: {
      minWidth: 200,
      maxWidth: `calc(100% - ${theme.spacing(7)})`,
      maxHeight: theme.mixins.toolbar.minHeight,
      color: grey[900]
    },
    select: {
      color: grey[900]
    }
  })

export interface SceneSearchProps extends WithStyles<typeof styles> {
  placeholder: string
}

function SceneSearch (props: SceneSearchProps) {
  const dispatch = useAppDispatch()
  const filters = useAppSelector(selectScenePickerFilters())
  const options = filters.map((filter) => {
    return { label: filter, value: filter }
  })

  const [searchInput, setSearchInput] = useState('')
  const handleInputChange = (searchInput: string) => {
    setSearchInput(searchInput)
  }

  const handleChange = (search: [{ label: string, value: string }]) => {
    const filters: string[] = []
    if (search != null) {
      for (const s of search) {
        if (
          ((s.value.startsWith('"') || s.value.startsWith('-"')) &&
            s.value.endsWith('"')) ||
          ((s.value.startsWith("'") || s.value.startsWith("-'")) &&
            s.value.endsWith("'"))
        ) {
          filters.push(s.value)
        } else {
          filters.push(...s.value.split(' '))
        }
      }
    }

    dispatch(setScenePickerFilters(filters))
  }

  const classes = props.classes
  return (
    <CreatableSelect
      className={clsx(classes.searchSelect, 'CreatableSelect')}
      components={{ DropdownIndicator: null }}
      value={options}
      options={options}
      inputValue={searchInput}
      isClearable
      isMulti
      rightAligned
      placeholder={props.placeholder}
      formatCreateLabel={(input: string) => 'Search for ' + input}
      onChange={handleChange}
      onInputChange={handleInputChange}
    />
  )
}

;(SceneSearch as any).displayName = 'SceneSearch'
export default withStyles(styles)(SceneSearch as any)
