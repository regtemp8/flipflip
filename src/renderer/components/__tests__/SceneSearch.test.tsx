import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import SceneSearch from "../SceneSearch";
import TestProvider from "../../../../test/util/TestProvider";
import store from "../../../store/store";

// TODO create functional tests instead of snapshots
describe("SceneSearch", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider store={store}>
        <SceneSearch placeholder='Search...' />
      </TestProvider>
    );

    let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
