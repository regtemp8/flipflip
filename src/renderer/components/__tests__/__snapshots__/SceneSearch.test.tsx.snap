// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneSearch should match snapshot 1`] = `
<div
  className="SceneSearch-searchSelect-1 CreatableSelect css-b62m3t-container"
  onKeyDown={[Function]}
>
  <span
    className="css-1f43avz-a11yText-A11yText"
    id="react-select-2-live-region"
  />
  <span
    aria-atomic="false"
    aria-live="polite"
    aria-relevant="additions text"
    className="css-1f43avz-a11yText-A11yText"
  />
  <div
    className=" css-1s2u09g-control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className=" css-319lph-ValueContainer"
    >
      <div
        className=" css-14el2xx-placeholder"
        id="react-select-2-placeholder"
      >
        Search...
      </div>
      <div
        className=" css-6j8wv5-Input"
        data-value=""
      >
        <input
          aria-autocomplete="list"
          aria-describedby="react-select-2-placeholder"
          aria-expanded={false}
          aria-haspopup={true}
          autoCapitalize="none"
          autoComplete="off"
          autoCorrect="off"
          className=""
          disabled={false}
          id="react-select-2-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          role="combobox"
          spellCheck="false"
          style={
            {
              "background": 0,
              "border": 0,
              "color": "inherit",
              "font": "inherit",
              "gridArea": "1 / 2",
              "label": "input",
              "margin": 0,
              "minWidth": "2px",
              "opacity": 1,
              "outline": 0,
              "padding": 0,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className=" css-1hb7zxy-IndicatorsContainer"
    />
  </div>
</div>
`;
